### PR TITLE
WIP: Add type annotations

### PR DESCRIPTION
**Pull request by FlorianMF**

### CODE REVIEW

File: torchserve_dashboard/api.py [hunk 1]
```diff
@@ -1,59 +1,73 @@
...
+            self.log_location, "torchserve_dashboard.log"
+        ) if self.log_location is not None else None
         torchserve_cmd = f"torchserve --start --ncs --model-store {self.model_store} --ts-config {self.config_path}"
         p = subprocess.Popen(
             torchserve_cmd.split(" "),
             env=self.env,
             stdout=subprocess.DEVNULL,
-            stderr=open(dashboard_log_path, "a+"),
+            stderr=open(dashboard_log_path, "a+")
+            if dashboard_log_path else None,
```
Comment (cceyda) [inline torchserve_dashboard/api.py:70]: 👍

File: torchserve_dashboard/api.py [hunk 2]
```diff
@@ -1,59 +1,73 @@
...
+            return "", e
+
+    def start_torchserve(self) -> str:
 
         if not os.path.exists(self.model_store):
             return "Can't find model store path"
         elif not os.path.exists(self.config_path):
             return "Can't find configuration path"
-        dashboard_log_path = os.path.join(self.log_location, "torchserve_dashboard.log")
+        dashboard_log_path: Optional[str] = os.path.join(
```
Comment (cceyda) [inline torchserve_dashboard/api.py:61]: Although I like type annotations for function input,outputs I find it slightly unnecessary to have typed inline(?) variables, at least for a project this size. 
IMO less is more at this point (I might be wrong)

File: torchserve_dashboard/dash.py [hunk 3]
```diff
@@ -41,14 +42,13 @@
 
 def check_args(args):
     M_API = "http://127.0.0.1:8081"
-    model_store = args.model_store
-    config_path = args.config_path
-    log_location = args.log_location
-    metrics_location = args.metrics_location
+    model_store: str = args.model_store
```
Comment (cceyda) [inline torchserve_dashboard/dash.py:45]: same here

File: torchserve_dashboard/api.py (hunk 2 above)
Comment (FlorianMF) [reply torchserve_dashboard/api.py:61]: I personally don't have a problem with inline type annotations. This is mainly due to the used libraries not providing types.
But I have to admit that I mainly use inline type annotations for the VSCode type checker to be happy ^^.
We can skip these if you want. Feel free to remove them and push the changes :)

**Pull request merged**